### PR TITLE
Point datatype

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/value.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ typedef enum {
 #define SI_TYPE(value) (value).type
 #define SI_NUMERIC (T_INT64 | T_DOUBLE)
 #define SI_GRAPHENTITY (T_NODE | T_EDGE)
-#define SI_ALL (T_MAP | T_NODE | T_EDGE | T_ARRAY | T_PATH | T_DATETIME | T_LOCALDATETIME | T_DATE | T_TIME | T_LOCALTIME | T_DURATION | T_STRING | T_BOOL | T_INT64 | T_DOUBLE | T_NULL | T_PTR)
+#define SI_ALL (T_MAP | T_NODE | T_EDGE | T_ARRAY | T_PATH | T_DATETIME | T_LOCALDATETIME | T_DATE | T_TIME | T_LOCALTIME | T_DURATION | T_STRING | T_BOOL | T_INT64 | T_DOUBLE | T_NULL | T_PTR | T_POINT)
 #define SI_VALID_PROPERTY_VALUE (T_POINT | T_ARRAY | T_DATETIME | T_LOCALDATETIME | T_DATE | T_TIME | T_LOCALTIME | T_DURATION | T_STRING | T_BOOL | T_INT64 | T_DOUBLE)
 #define SI_INDEXABLE (SI_NUMERIC | T_BOOL | T_STRING)
```
Comment (jeffreylovitz) [inline src/value.h:55]: Shouldn't `T_POINT` be included in this bitmap, or is it deliberately excluded?

File: src/value.c [hunk 2]
```diff
@@ -527,6 +531,13 @@ int SIValue_Compare(const SIValue a, const SIValue b, int *disjointOrNull) {
 			return Map_Compare(a, b, disjointOrNull);
 		case T_NULL:
 			break;
+		case T_POINT:
+		{
+			float lon_diff = SAFE_COMPARISON_RESULT(Point_lon(a) - Point_lon(b));
```
Comment (jeffreylovitz) [inline src/value.c:536]: Should be an int, `SAFE_COMPARISON_RESULT` will return -1, 0, or 1.

File: tests/flow/test_point.py [hunk 3]
```diff
@@ -143,3 +143,45 @@ def test_point_index_lookup(self):
...
+        q = "MATCH (n:L) SET n.v = [point({latitude:34, longitude:35})] RETURN n.v"
+        res = redis_graph.query(q)
+        self.env.assertEquals(res.result_set[0][0], expected_value)
+
+        # order a list of points
+        expected_value = [
+        {'latitude':32, 'longitude':31},
+        {'latitude':32, 'longitude':32},
+        {'latitude':33, 'longitude':35},
+        {'latitude':29, 'longitude':36}
```
Comment (jeffreylovitz) [inline tests/flow/test_point.py:170]: Use two points with the same longitude value to test latitude comparison (will suppress Codecov complaint).

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
This looks great, I have only a few minor notes.

File: src/value.h (hunk 1 above)
Comment (swilly22) [reply src/value.h:55]: it should, currently this is deliberate.

**Review by swilly22 — APPROVED** [review]

**Pull request merged**